src/core/libmaven/csvreports.cpp

```diff
@@ -365,9 +366,21 @@ void CSVReports::writePeakInfo(PeakGroup* group) {
         if(group->label !='g') return;
     } else if (selectionFlag == 3) {
         if(group->label !='b') return;
-    } else {
-
     }
+
+    // sort the peaks in the group according to the sample names using a lambda function.
+    // this ensures that the order in which the peaks are written is same across different systems.
+    std::sort(group->peaks.begin(), group->peaks.end(), [](Peak& lhs, Peak& rhs) {
```

Comment:
@saifulbkhan 
This method already exists in Peak class. 
`bool Peak::compSampleName(const Peak& a, const Peak& b)`
You can just use it like this:
`sort(group->peaks.begin(), groups->peaks.end(), Peak::compSampleName)`;

Reply:
@shubhra-agrawal 
This seems quite apparent, but by any chance could a peak not have a sample related to it? The [comparison function](https://github.com/ElucidataInc/ElMaven/blob/develop/src/core/libmaven/Peak.cpp#L236) in Peak class does not check for such a case but before writing peak values CSVReports class does [check](https://github.com/ElucidataInc/ElMaven/blob/develop/src/core/libmaven/csvreports.cpp#L375) if the corresponding sample is not `NULL`.

Either way I should be using the Peak class' given comparator instead of my own. Just wanted to make sure that this comparator is not missing an odd edge case.

Reply:
@saifulbkhan every peak should have a corresponding sampleName. If it's not there, that in itself tells us something's wrong.

Reply:
@shubhra-agrawal cool. Already made the changes you requested btw.